### PR TITLE
generic role assignment mechanism for all entity types.

### PR DESCRIPTION
**Pull request by PawelPawelec-RDX**

### CODE REVIEW

File: src/RadixDlt.NetworkGateway.PostgresIntegration/Services/BlueprintProvider.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    )
+    {
+        var blueprintNames = blueprintDefinitions.Select(x => x.Name).ToList();
+        var blueprintVersions = blueprintDefinitions.Select(x => x.Version).ToList();
+        var packageEntityIds = blueprintDefinitions.Select(x => x.PackageEntityId).ToList();
+
+        var result = await _dbContext
+            .PackageBlueprintHistory
+            .FromSqlInterpolated($@"
+WITH blueprints (blueprint_name, blueprint_version, package_entity_id) AS (SELECT UNNEST({blueprintNames}), UNNEST({blueprintVersions}), UNNEST({packageEntityIds}))
```
Comment (krzlabrdx) [inline src/RadixDlt.NetworkGateway.PostgresIntegration/Services/BlueprintProvider.cs:41]: `WITH variables` to stay consistent?

File: src/RadixDlt.NetworkGateway.PostgresIntegration/Services/BlueprintProvider.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+SELECT pbh.*
+FROM blueprints b
+INNER JOIN package_blueprint_history pbh
+ON pbh.name = b.blueprint_name AND pbh.version = b.blueprint_version and pbh.package_entity_id = b.package_entity_id
```
Comment (krzlabrdx) [inline src/RadixDlt.NetworkGateway.PostgresIntegration/Services/BlueprintProvider.cs:45]: Right now I believe we have index on `package_entity_id, from_state_version` only, is it enough?

File: src/RadixDlt.NetworkGateway.PostgresIntegration/Services/RoleAssignmentsKeyProvider.cs [hunk 3]
```diff
@@ -62,64 +62,69 @@
...
-            .Concat(GetKeysWithUpdaterRoles(_fungibleMainModuleResourceRuleKeys, ObjectModuleId.Main))
-            .ToList();
     }
 
     public List<RoleAssignmentEntry> GetNativeModulesKeys() => _nativeModulesKeys;
 
-    public List<RoleAssignmentEntry> GetNonFungibleResourceKeys() => _nonFungibleResourceKeys;
-
-    public List<RoleAssignmentEntry> GetFungibleResourceKeys() => _fungibleResourceKeys;
+    public List<RoleAssignmentEntry> ExtractKeysFromBlueprintAuthConfig(CoreModel.AuthConfig authConfig)
```
Comment (krzlabrdx) [inline src/RadixDlt.NetworkGateway.PostgresIntegration/Services/RoleAssignmentsKeyProvider.cs:111]: How about `return authConfig.MethodRoles?.Roles.Select(...).ToList() ?? new List<Role...>`?


1. No null negations.
2. Ideally we might want to define `private static List<...> Empty` to reduce GC pressure.

**Review by krzlabrdx — CHANGES_REQUESTED** [review]

File: src/RadixDlt.NetworkGateway.PostgresIntegration/Services/BlueprintProvider.cs (hunk 1 above)
Comment (PawelPawelec-RDX) [reply src/RadixDlt.NetworkGateway.PostgresIntegration/Services/BlueprintProvider.cs]: sure, done.

File: src/RadixDlt.NetworkGateway.PostgresIntegration/Services/BlueprintProvider.cs (hunk 2 above)
Comment (PawelPawelec-RDX) [reply src/RadixDlt.NetworkGateway.PostgresIntegration/Services/BlueprintProvider.cs]: good catch, added new index.

File: src/RadixDlt.NetworkGateway.PostgresIntegration/CommonDbContext.cs [hunk 4]
```diff
@@ -416,6 +416,10 @@ private static void HookupHistory(ModelBuilder modelBuilder)
             .Entity<PackageBlueprintHistory>()
             .HasIndex(e => new { e.PackageEntityId, e.FromStateVersion });
 
+        modelBuilder
+            .Entity<PackageBlueprintHistory>()
+            .HasIndex(e => new { e.Name, e.Version, e.PackageEntityId, e.FromStateVersion });
```
Comment (krzlabrdx) [inline src/RadixDlt.NetworkGateway.PostgresIntegration/CommonDbContext.cs:421]: Guess we might want to define it as `PackageEntityId, Name, Version, FromStateVersion`.

**Review by krzlabrdx — APPROVED** [review]

**Pull request merged**